sources/packages/backend/libs/integrations/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -27,23 +27,52 @@ export class StudentAssessmentService extends RecordDataModelService<StudentAsse
     return this.repo.find({
       select: {
         id: true,
+        assessmentData: true as unknown,
+        workflowData: true as unknown,
+        assessmentDate: true,
+        triggerType: true,
         application: {
+          id: true,
           applicationNumber: true,
+          data: true as unknown,
+          submittedDate: true,
+          applicationStatus: true,
+          applicationStatusUpdatedOn: true,
           student: {
-            sinValidation: { sin: true },
-            user: { lastName: true, firstName: true },
+            id: true,
+            sinValidation: { id: true, sin: true },
+            user: { id: true, lastName: true, firstName: true },
             birthDate: true,
+            contactInfo: true as unknown,
+            studentRestrictions: {
+              id: true,
+              isActive: true,
+              restriction: { id: true, restrictionCode: true },
```

Comment:
is it possible to add a filter in the where to select only the restrictions that need to be checked? Only active ones, for instance.

Reply:
As discussed, tested with where condition added, it worked fine for a student who does not have a restriction and who has at least one  active restriction, but if the student has restrictions which are not active(resolved) then any assessment for that student is not returned.

Hence I changed the code back to what it was.

Reply:
Thanks for putting the effort into it. As we discussed lest put an effort during the upcoming tickets to have this query possibility changed to a typeorm "string query" that would allow it.